### PR TITLE
DPL Analysis: implementation of expression columns with new functions

### PR DESCRIPTION
**Pull request by aalkin**
WARNING: will not work without https://github.com/alisw/alidist/pull/2343

* New functions for expressions
* p, pT, eta and phi as expression columns (currently phi is not normalized to [0:2pi), use dynamic nphi to get that)
*  Adapted filters.cxx example
* Adapted histograms.cxx example

### CODE REVIEW

File: Analysis/Tasks/jetfinder.cxx [hunk 1]
```diff
@@ -96,7 +94,7 @@ struct JetFinderTask {
 
     std::vector<fastjet::PseudoJet> inputParticles;
     for (auto& track : fullTracks) {
-      auto energy = std::sqrt(track.p2() + mPionSquared);
+      auto energy = std::sqrt(track.p() * track.p() + mPionSquared);
```
Comment (ktf) [inline Analysis/Tasks/jetfinder.cxx:97]: I assume p() has a sqrt inside? does it really make sense to have it rather than just p2?

Comment (aalkin) [reply Analysis/Tasks/jetfinder.cxx:97]: I've tried to keep data model as it was before. We can redefine what we want calculated after people have "played" with new capabilities.

Comment (ktf) [reply Analysis/Tasks/jetfinder.cxx:97]: ok, but here the old version was using p2(), why changing it then? Anyways, fine for me.

Comment (aalkin) [reply Analysis/Tasks/jetfinder.cxx:97]: 'p2' was from the implementation with no functions. So far it was used only in this file, it seems.

**Comment by ktf** [comment]
k

File: Analysis/Tasks/jetfinder.cxx [hunk 2]
```diff
@@ -64,11 +64,9 @@ struct JetFinderTask {
   fastjet::AreaDefinition areaDefBkg{areaType, ghostSpec};
   fastjet::Selector selBkg = fastjet::SelectorAbsRapMax(rapBkg);
 
-  // TODO: use abs, eventually use pt
   // TODO: use values from configurables
   // TODO: add eta cuts
-  Filter trackCuts = (aod::track::signed1Pt < 10.f) &&
-                     (aod::track::signed1Pt > -10.f);
+  Filter trackCuts = nabs(aod::track::pt) > 0.1f;
```
Comment (jgrosseo) [inline Analysis/Tasks/jetfinder.cxx:70]: Shouldn't pt be always positive anyway?

File: Framework/Core/include/Framework/AnalysisDataModel.h [hunk 3]
```diff
@@ -83,17 +83,20 @@ DECLARE_SOA_COLUMN(Z, z, float);
 DECLARE_SOA_COLUMN(Snp, snp, float);
 DECLARE_SOA_COLUMN(Tgl, tgl, float);
 DECLARE_SOA_COLUMN(Signed1Pt, signed1Pt, float);
-DECLARE_SOA_DYNAMIC_COLUMN(Phi, phi, [](float snp, float alpha) -> float {
-  float phi = std::asin(snp) + alpha;
-  constexpr float twopi = 2.0 * M_PI;
+// FIXME: find a way to clean up node templates so '1.f *' workaround is not needed
+DECLARE_SOA_EXPRESSION_COLUMN(Phi, phi, float, nasin(aod::track::snp) + aod::track::alpha);
```
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h:87]: What you have as 'nphi' must be the default. So already renormalized within 0 and 2 pi. 
You can just rename:
phi --> phiraw
nphi --> phi

File: Analysis/Tasks/jetfinder.cxx (hunk 2 above)
Comment (aalkin) [reply Analysis/Tasks/jetfinder.cxx:70]: You are right, I'll fix it as soon as tests finish.

File: Framework/Core/include/Framework/AnalysisDataModel.h (hunk 3 above)
Comment (aalkin) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: Done.

File: Analysis/Tasks/jetfinder.cxx (hunk 2 above)
Comment (aalkin) [reply Analysis/Tasks/jetfinder.cxx:70]: Done.

**Comment by ktf** [comment]
@aalkin error is unrelated. Merging.

@sawenzel there seems to be some issues with one of the Geant4 tests. Did anything change there?

**Pull request merged**